pkg/catalogv2/git/git.go

```diff
@@ -1,480 +1,440 @@
+// or an error if any step in this process fails.
+func BuildRepoConfig(secret *corev1.Secret, namespace, name, gitURL string, insecureSkipTLS bool, caBundle []byte) (*repository, error) {
+	isGitSSH, err := isGitSSH(gitURL)
+	if err != nil {
+		return nil, fmt.Errorf("failed to verify the type of URL %s: %w", gitURL, err)
 	}
-
-	output := &bytes.Buffer{}
-	if err := g.gitCmd(output, "ls-remote", "--", g.URL, formatRefForBranch(branch)); err != nil {
-		return "", err
+	if !isGitSSH {
+		u, err := url.Parse(gitURL)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse URL %s: %w", gitURL, err)
+		}
+		if u.Scheme != "http" && u.Scheme != "https" {
+			return nil, fmt.Errorf("invalid git URL scheme %s, only http(s) supported", u.Scheme)
+		}
 	}
-
-	var lines []string
-	s := bufio.NewScanner(output)
-	for s.Scan() {
-		lines = append(lines, s.Text())
+	dir := gitDir(namespace, name, gitURL)
+	headers := map[string]string{}
+	if settings.InstallUUID.Get() != "" {
+		headers["X-Install-Uuid"] = settings.InstallUUID.Get()
+	}
+	// convert caBundle to PEM format because git requires correct line breaks, header and footer.
+	if len(caBundle) > 0 {
+		caBundle = convertDERToPEM(caBundle)
+		insecureSkipTLS = false
+	}
+
+	repo := &repository{
+		URL:               gitURL,
+		Directory:         dir,
+		caBundle:          caBundle,
+		insecureTLSVerify: insecureSkipTLS,
+		secret:            secret,
+		headers:           headers,
+		repoGogit:         &gogit.Repository{},
+		cloneOpts:         &gogit.CloneOptions{},
+		checkoutOpts:      &gogit.CheckoutOptions{},
+		fetchOpts:         &gogit.FetchOptions{},
+		listOpts:          &gogit.ListOptions{},
+		resetOpts:         &gogit.ResetOptions{},
+	}
+
+	// credentials must be setted before options
+	err = repo.setRepoCredentials()
+	if err != nil {
+		return repo, err
 	}
+	repo.setRepoOptions()
 
-	return firstField(lines, fmt.Sprintf("no commit for branch: %s", branch))
+	return repo, nil
 }
 
-// Head runs git clone on directory(if not exist), reset dirty content and return the HEAD commit
-func (g *git) Head(branch string) (string, error) {
-	if err := g.clone(branch); err != nil {
-		return "", err
+// setRepoCredentials detects which type of authentication and communication protocol
+// and configurates the git repo communications accordingly
+func (r *repository) setRepoCredentials() error {
+	if r.secret == nil {
+		return nil
 	}
 
-	if err := g.reset("HEAD"); err != nil {
-		return "", err
-	}
+	var username string
+	var password string
 
-	return g.currentCommit()
-}
+	switch r.secret.Type {
+	case corev1.SecretTypeBasicAuth: // BASIC HTTP(S) AUTHENTICATION
+		// get the credentials setted in kubernetes
+		username = string(r.secret.Data[corev1.BasicAuthUsernameKey])
+		password = string(r.secret.Data[corev1.BasicAuthPasswordKey])
+		if len(password) == 0 || len(username) == 0 {
+			return fmt.Errorf("username and password not provided")
```

Comment:
nit: username or password not provided

Reply:
I have implemented a fix, please resolve the conversation